lib/controllers/messages.ts

```diff
@@ -606,15 +615,31 @@ export function GetMessageHandler(
         }|${errorOrNotificationStatuses.value.message}`
       );
     }
-
     const notificationStatuses = errorOrNotificationStatuses.value;
 
-    const messageStatus: MessageResponseWithContent = {
+    const errorOrMaybeMessageStatus = await messageStatusModel.findOneByMessageId(
+      retrievedMessage.id
+    );
+
+    if (isLeft(errorOrMaybeMessageStatus)) {
+      return ResponseErrorInternal(
+        `Error retrieving MessageStatus: ${
+          errorOrMaybeMessageStatus.value.code
+        }|${errorOrMaybeMessageStatus.value.body}`
+      );
+    }
+    const maybeMessageStatus = errorOrMaybeMessageStatus.value;
+
+    const returnedMessage: MessageResponseWithContent = {
       message,
-      notification: notificationStatuses.toUndefined()
+      notification: notificationStatuses.toUndefined(),
+      // we do not return the status date-time
+      status: maybeMessageStatus
+        .map(messageStatus => messageStatus.status)
+        .getOrElse(MessageStatusValueEnum.ACCEPTED)
     };
 
-    return ResponseSuccessJson(messageStatus);
+    return ResponseSuccessJson(withoutUndefinedValues(returnedMessage));
```

Comment:
split this in two steps and explain why `withoutUndefinedValues` is needed

Reply:
I've removed it (it's not really needed, it was just to avoid return { email: undefined } for the notification statuses)